### PR TITLE
Simple cpicker improvements

### PR DESCRIPTION
**Pull request by paulpv**
1. Consolidate separate h, s, & v vars to single lv_color_hsv_t
2. Emit VALUE_CHANGED when reset

### CODE REVIEW

File: src/lv_objx/lv_cpicker.c [hunk 1]
```diff
@@ -1279,83 +1303,40 @@ static lv_res_t lv_cpicker_signal(lv_obj_t * cpicker, lv_signal_t sign, void * p
...
         case LV_CPICKER_COLOR_MODE_VALUE:
-            ext->value = LV_CPICKER_DEF_VALUE;
-            ext->prev_value = ext->value;
+            ext->hsv.v = LV_CPICKER_DEF_HSV.v;
             break;
         }
+        ext->prev_hsv = ext->hsv;
         lv_cpicker_invalidate(cpicker, false);
+
+        res = lv_event_send(cpicker, LV_EVENT_VALUE_CHANGED, NULL);
```
Comment (embeddedt) [inline src/lv_objx/lv_cpicker.c:1320]: Technically, AFAIK, the color value hasn't really changed. The `LV_EVENT_VALUE_CHANGED` is really to convey changes in the value (for controls like sliders/switches).

Comment (paulpv) [reply src/lv_objx/lv_cpicker.c:1320]: I can add a test to only emit if the value wasn't the default

Comment (embeddedt) [reply src/lv_objx/lv_cpicker.c:1320]: I was confused by your message so I decided to read through the whole function. Now I see that it is actually changing the color value. Sorry about that. Nothing needs to be changed here.

**Comment by embeddedt** [comment]
While we have this PR open, in `lv_cpicker_signal`, I see that there is a duplicated code path for the arrow keys. The left and up arrow keys appear to be handled the exact same way. The same goes for the right and down arrow keys. Can this be consolidated?

**Comment by paulpv** [comment]
Can do.
Didn't at first because the code path would get a little ugly.
I'll improve it.

**Comment by paulpv** [comment]
@embeddedt regarding the arrow keys code path, the arrow handling code is already consolidated circa line 1201.
Or are you talking about something else?

**Comment by embeddedt** [comment]
> Or are you talking about something else?

I was talking about what you consolidated in https://github.com/littlevgl/lvgl/pull/1208/commits/bf9ed06bac67baf97127102362be5e62d2348f6a. It's fixed now. :slightly_smiling_face:

**Comment by kisvegabor** [comment]
It can be merged from my side :slightly_smiling_face:

**Pull request merged**